bokehjs/src/coffee/common/embed.coffee

```diff
@@ -8,6 +8,18 @@ HasProperties = require "./has_properties"
 {pull_session} = require "./client"
 {Promise} = require "es6-promise"
 
+_handle_notebook_comms = (msg) ->
+  logger.debug("handling notebook comms")
+  # @ is bound to the doc
+  @apply_json_patch_string(msg.content.data)
+
+_init_comms = (target, doc) ->
+  logger.info("Initializing Jupyter comms for target #{target} and doc #{doc}")
+  comm_manager = Jupyter.notebook.kernel.comm_manager
```

Comment:
But that does not work now? Just wondering if there is any path that leads to "future proof".  `define` is synchronous at least, right? Maybe something like this could work:
```
if Jupyter?
    comm_manager = Jupyter.notebook.kernel.comm_manager
    comm_manager.register_target(target, (comm, msg) ->
        comm.on_msg(_.bind(_handle_notebook_comms, doc))
    )
else 
    define([ 'base/js/namespace'], (Jupyter) ->
        comm_manager = Jupyter.notebook.kernel.comm_manager
        comm_manager.register_target(target, (comm, msg) ->
            comm.on_msg(_.bind(_handle_notebook_comms, doc))
        )
```